src/Simplify_Stmts.cpp

```diff
@@ -448,6 +447,25 @@ Stmt Simplify::visit(const Block *op) {
             result = Block::make(result, if_rest);
         }
         return result;
+    } else if (if_first &&
```

Comment:
Is this strictly *necessary* or just a drive-by optimization? (Wondering if this could be landed separately to try to flush out the weird failure)

Reply:
These two simplifications when combined are what turns:
```
if (x == 0) {
} 
if (x == 1) {
}
```
into
```
if (x == 0) {
} else if (x == 1) {
}
```

which is the pattern that codegen expects. 

It could be landed separately I suppose, but it's part of the same change, not a drive-by